### PR TITLE
Add monitoring dashboard to fdsc-dashboard

### PR DESCRIPTION
**Pull request by wistefan**
* adds prometheus to collect metrics from apisix
* enable apisix-prometheus plugin
* add grafana and default dashboard for routes
* embedd dashboard in fdsc-dashboard
* set cert-manager alt-names annotation to fix traefik cert delivery

### CODE REVIEW

**Review by vramperez — COMMENTED** [review]
In general, it looks good to me, but I think adding monitoring to the k3s deployments for the tests will only make them heavier and cause issues when deploying locally; plus, none of that is actually used in the ITs

**Comment by wistefan** [comment]
> In general, it looks good to me, but I think adding monitoring to the k3s deployments for the tests will only make them heavier and cause issues when deploying locally; plus, none of that is actually used in the ITs

You are right. I will extract it into a dedicated profile.

File: charts/data-space-connector/templates/apisix-prometheus-global-rule.yaml
Comment (Mortega5) [inline charts/data-space-connector/templates/apisix-prometheus-global-rule.yaml:1]: I think this is already covered by apisix chart with a [single flag ](https://github.com/apache/apisix-helm-chart/blob/master/charts/apisix/values.yaml#L560). Is this needed?

File: charts/data-space-connector/values.yaml [hunk 1]
```diff
@@ -130,6 +130,24 @@ decentralizedIam:
...
+        prometheus.io/port: "9091"
+      apisix:
+        prometheus:
+          enabled: true
+
+      plugins:
+        - prometheus
+
+      pluginAttrs:
+        prometheus:
```
Comment (Mortega5) [inline charts/data-space-connector/values.yaml]: this is set automatically by the [apisix chart](https://github.com/apache/apisix-helm-chart/blob/master/charts/apisix/templates/_helpers.tpl#L83) when prometheus is enabled

File: charts/data-space-connector/values.yaml [hunk 2]
```diff
@@ -2222,4 +2240,10 @@ extraManifests: []
 
 # -- dashboard for operating the FIWARE Data Space Connector
 fdsc-dashboard:
+  enabled: false
+
+prometheus:
```
Comment (Mortega5) [inline charts/data-space-connector/values.yaml:2268]: I think grafana and prometheus should not be part of this chart. In fact, it is usually used as a [kube-stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack#kube-prometheus-stack) rather than separately. In this case, prometheus.io annotations can be replaced by setting [`metrics.serviceMonitor.enabled`](https://github.com/apache/apisix-helm-chart/blob/master/charts/apisix/values.yaml#L266)

File: charts/data-space-connector/templates/grafana-dashboard-apisix.yaml
Comment (Mortega5) [inline charts/data-space-connector/templates/grafana-dashboard-apisix.yaml:1]: if we remove grafana dependency, we should move this to apisix section. Something like
```yaml
# [...]
apisix:
  grafana:
    dashboard:
      enabled: true
      labels: 
        grafana_dashboard: "1"
      annotations:
        grafana_folder: dsc
```

File: charts/data-space-connector/values.yaml (hunk 2 above)
Comment (wistefan) [reply charts/data-space-connector/values.yaml:2268]: I think this is a similar case as we have with the databases. Since we have features that require grafana/prometheus/mongodb etc. we provided them in a default setup, but its possible to use existing solutions of the cluster. Thats why it should be part of the chart in my opinion, just disabled by default.

File: charts/data-space-connector/templates/grafana-dashboard-apisix.yaml
Comment (wistefan) [reply charts/data-space-connector/templates/grafana-dashboard-apisix.yaml:1]: I dont agree. Grafana is not purely for apisix here. Yes, the main source of data is apisix, but grafana will also be used for visualizing tracing infromation. I dont see why this should be part of the apisix config.

**Review by Mortega5 — APPROVED** [review]

**Review by vramperez — APPROVED** [review]
